### PR TITLE
Use Reactive Streams Cassandra API in 4.5.0

### PR DESCRIPTION
**Pull request by ennru**
* Upgrade to Datastax Java Driver 4.4.0
* Use Reactive Streams publisher API
* Clean up `CassandraSession` (commit 2)

### CODE REVIEW

File: session/src/main/scala/akka/stream/alpakka/cassandra/scaladsl/CassandraSession.scala [hunk 1]
```diff
@@ -301,68 +275,13 @@ final class CassandraSession(
...
   def selectOne(stmt: String, bindValues: AnyRef*): Future[Option[Row]] = {
-    val bound: Future[BoundStatement] = prepare(stmt).map(
-      ps =>
-        if (bindValues.isEmpty) ps.bind()
-        else ps.bind(bindValues: _*))
-    bound.flatMap(bs => selectOne(bs))
+    bind(stmt, bindValues).flatMap(bs => selectOne(bs))
   }
 
-  private class SelectSource(stmt: Future[Statement[_]]) extends GraphStage[SourceShape[Row]] {
```
Comment (patriknw) [inline session/src/main/scala/akka/stream/alpakka/cassandra/scaladsl/CassandraSession.scala]: We should do some performance benchmark of this change. Great to get rid of this, but we should know that it doesn't reduce performance. For example the Publisher will add an async boundary for each emitted row while this custom GraphStage will be fused with the rest of the stream.

**Comment by chbatey** [comment]
Great that this made it back into the cassandra driver and no the commercial one

File: core/src/main/resources/reference.conf [hunk 2]
```diff
@@ -478,6 +478,8 @@ akka.persistence.cassandra {
 
 
 //#profile
+# See reference configuration at
+# https://docs.datastax.com/en/developer/java-driver/4.4/manual/core/configuration/reference/
```
Comment (patriknw) [inline core/src/main/resources/reference.conf]: 4.5 now

**Comment by ennru** [comment]
We still need to understand if the test failures #469 that started to show again are related to the later version of the driver.
I made this PR contain the new test case from #717 which is meant to be used to identify changes in performance when comparing the Akka Streams source implementation with using the new driver-native Reactive Streams API.

**Comment by ennru** [comment]
The performance drop was considered to be real and fixed with #730.

**Pull request merged**